### PR TITLE
Kubernetes intro

### PR DESCRIPTION
**Pull request by drozdnikita**
# Выполнено ДЗ № 1

 - [x] Основное ДЗ
 - [x] Задание со *

### PODs self-restoration:

1. the desired number of coredns pods is described via ReplicaSet coredns-f9fd979d6, thus once the pod gets killed it gets restored immediately
2. etcd-minikube, kube-apiserver-minikube, kube-controller-manager-minikube, kube-scheduler-minikube are part on the node and described as Non-terminated pods there
3. the desired number of kube-proxy pods is described via DaemonSet kube-proxy, and the same as with coredns gets restored right away

### Dockerfile

busybox as a base image running httpd via CMD. The resulting docker image is drozdnikita/drozd_web_app:1.0.0.

### web pod yaml

works as expected, nothing to add really

### healthy frontend
docker image **drozdnikita/drozd_frontend:1.0.0** to be used
also it required variables to be added in the 'env' of pod manifest

## Как запустить проект:
**web**
in drozdnikita_platform/kubernetes-intro run the command 
`kubectl apply -f web-pod.yaml`
`kubectl port-forward --address 0.0.0.0 pod/web 8000:8000`

**frontend**
in drozdnikita_platform/kubernetes-intro run the command 
`kubectl apply -f frontend-pod-healthy.yaml`

## Как проверить работоспособность:
**web**
open the link [http://localhost:8000/index.html](http://localhost:8000/index.html)
**frontend**
_kubectl logs frontend_ shows no errors but something like
`kubectl logs frontend`
`{"message":"Tracing enabled.","severity":"info","timestamp":"2021-01-05T13:14:22.36757906Z"}`
`{"message":"Profiling enabled.","severity":"info","timestamp":"2021-01-05T13:14:22.367633283Z"}`
`{"message":"jaeger initialization disabled.","severity":"info","timestamp":"2021-01-05T13:14:22.367719165Z"}`

## PR checklist:
 - [x] Выставлен label с темой домашнего задания

### CODE REVIEW

File: kubernetes-intro/web-pod.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    command: ['sh', '-c', 'wget -O- https://tinyurl.com/otus-k8s-intro | sh']
+    volumeMounts:
+    - name: app
+      mountPath: /app
+  volumes:
+  - name: app
+    emptyDir: {}
+  - name: web-drozd
+  containers:
+    image: drozdnikita/drozd_web_app:1.0.0
```
Comment (Jasstkn) [inline kubernetes-intro/web-pod.yaml:21]: контейнеру нужно имя, иначе под не задеплоится

**Review by Jasstkn — CHANGES_REQUESTED** [review]
Добрый день. Оставила комментарий, что нужно поправить. В остальном все сделано верно, как только внесете изменение - приму домашку

File: kubernetes-intro/web-pod.yaml (hunk 1 above)
Comment (drozdnikita) [reply kubernetes-intro/web-pod.yaml:21]: Добрый день,

исправил. проверьте, пожалуйста

**Review by Jasstkn — APPROVED** [review]
Approved! 🖖

**Pull request merged**